pkg/performanceprofile/profilecreator/profilecreator_test.go

```diff
@@ -404,6 +404,287 @@ var _ = Describe("PerformanceProfileCreator: Populating Reserved and Isolated CP
+			disableHT = false
+			mustGatherDirAbsolutePath, err = filepath.Abs(mustGatherDirPath)
+			Expect(err).ToNot(HaveOccurred())
+			handle, err = NewGHWHandler(mustGatherDirAbsolutePath, node)
+			Expect(err).ToNot(HaveOccurred())
+			_, _, _, err := handle.CalculateCPUSets(reservedCPUCount, offlinedCPUCount, splitReservedCPUsAcrossNUMA, disableHT)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("Errors out in case offlinedCPUCount specified is greater than the number of CPUs in system", func() {
+			reservedCPUCount = 20  // random number, no meaning
+			offlinedCPUCount = 100 // random number should be greater than the actual number of CPUs in must-gather
+			splitReservedCPUsAcrossNUMA = true
+			disableHT = false
+			mustGatherDirAbsolutePath, err = filepath.Abs(mustGatherDirPath)
+			Expect(err).ToNot(HaveOccurred())
+			handle, err = NewGHWHandler(mustGatherDirAbsolutePath, node)
+			Expect(err).ToNot(HaveOccurred())
+			_, _, _, err := handle.CalculateCPUSets(reservedCPUCount, offlinedCPUCount, splitReservedCPUsAcrossNUMA, disableHT)
+			Expect(err).To(HaveOccurred())
+		})
+		It("Everything is ok in case offlinedCPUCount plus reservedCPUCount specified is equal to the number of CPUs in system", func() {
+			reservedCPUCount = 20 // random number, no meaning
+			offlinedCPUCount = 60 // random number --> both `reservedCPUCount` plus `offlinedCPUCount` should sum 80 (which is the number of CPUs in the must-gather)
+			Expect(reservedCPUCount + offlinedCPUCount).To(Equal(80))
+			splitReservedCPUsAcrossNUMA = true
+			disableHT = false
+			mustGatherDirAbsolutePath, err = filepath.Abs(mustGatherDirPath)
+			Expect(err).ToNot(HaveOccurred())
+			handle, err = NewGHWHandler(mustGatherDirAbsolutePath, node)
+			Expect(err).ToNot(HaveOccurred())
+			_, _, _, err := handle.CalculateCPUSets(reservedCPUCount, offlinedCPUCount, splitReservedCPUsAcrossNUMA, disableHT)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("Ensure offlined CPUs populated are correctly when splitReservedCPUsAcrossNUMA is disabled, disableHT is disabled and we can offline a complete socket", func() {
+			reservedCPUCount = 20 // random number, no special meaning
+			offlinedCPUCount = 40 // NOT A random number: It is the number of CPUS in a socket
+			splitReservedCPUsAcrossNUMA = false
+			disableHT = false
+			mustGatherDirAbsolutePath, err = filepath.Abs(mustGatherDirPath)
+			Expect(err).ToNot(HaveOccurred())
+			handle, err = NewGHWHandler(mustGatherDirAbsolutePath, node)
+			Expect(err).ToNot(HaveOccurred())
+			reservedCPUSet, isolatedCPUSet, offlinedCPUSet, err := handle.CalculateCPUSets(reservedCPUCount, offlinedCPUCount, splitReservedCPUsAcrossNUMA, disableHT)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(offlinedCPUSet.Intersection(reservedCPUSet).IsEmpty()).To(BeTrue())
+			Expect(offlinedCPUSet.Intersection(isolatedCPUSet).IsEmpty()).To(BeTrue())
+			Expect(offlinedCPUSet.Size()).To(BeNumerically("==", offlinedCPUCount))
```

Comment:
we should probalby add a check that the union of the returned set is the total set of the CPUs in the system

Reply:
I have been trying to add this but reserved + offlined + isolated = total is not always true. 

When `disable-ht` is `true` and reserved are calculated topology is modified deleting all the "sibling" logical processors so they could not be eligible when isolated is calculated. 

This modified topology is used to calculate isolated but not for the offflined because in fact in this case we really want to offline the sibling logical processors.

TL;DR
So, if `disable-ht` is `true` then `reserved + offlined + isolated = total` will be `false` **unless** offlined-cpu-count >= number of logical processor siblings

Reply:
That's a great testcase to add to document this behaviour :) This kind of knowledge should be explicitely recorded, better in tests (+ explanation comments where helpful)

Reply:
done